base/dates/io.jl

```diff
@@ -71,7 +71,7 @@ function DateFormat(f::AbstractString,locale::AbstractString="english")
     for (i,k) in enumerate(s)
         k == "" && break
         tran = i >= endof(s) ? r"$" : match(Regex("(?<=$(s[i])).*(?=$(s[i+1]))"),f).match
-        slot = tran == "" || tran == r"$" ? FixedWidthSlot : DelimitedSlot
+        slot = tran == "" ? FixedWidthSlot : DelimitedSlot
```

Comment:
Yep, I think that's the case. The slot-parsing code here went through a few iterations, so my guess is that it may have been more relevant at an earlier iteration, but then didn't matter as much *and* didn't cause anything else to fail since it was always false.